### PR TITLE
chore: Update pre-commit checks

### PR DESCRIPTION
**Pull request by raffaellasuardini**
- updated .pre-commit-config.yaml: inserted ruff and pre-commit hooks

- created Ruff configuratio in pyproject.toml

### CODE REVIEW

File: djangocms_alias/constants.py [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-CATEGORY_LIST_URL_NAME = f'{PLUGIN_URL_NAME_PREFIX}_category_changelist'
-CHANGE_CATEGORY_URL_NAME = f'{PLUGIN_URL_NAME_PREFIX}_category_change'
-SELECT2_ALIAS_URL_NAME = f'{PLUGIN_URL_NAME_PREFIX}_select2'
-USAGE_ALIAS_URL_NAME = f'{PLUGIN_URL_NAME_PREFIX}_alias_usage'
-LIST_ALIAS_URL_NAME = f'{PLUGIN_URL_NAME_PREFIX}_alias_changelist'
-CHANGE_ALIAS_URL_NAME = f'{PLUGIN_URL_NAME_PREFIX}_alias_change'
-DELETE_ALIAS_URL_NAME = f'{PLUGIN_URL_NAME_PREFIX}_alias_delete'
-CATEGORY_SELECT2_URL_NAME = f'{PLUGIN_URL_NAME_PREFIX}_category_list_select2'
+CREATE_ALIAS_URL_NAME = f"{PLUGIN_URL_NAME_PREFIX}_create"
+DELETE_ALIAS_URL_NAME = f"{PLUGIN_URL_NAME_PREFIX}_delete"
```
[the remark this answers is not in the document]
Comment (marksweb) [reply djangocms_alias/constants.py:5]: This gets redefined further down, so can we delete `DELETE_ALIAS_URL_NAME` from line 5 please?

**Review by marksweb — CHANGES_REQUESTED** [review]

**Comment by marksweb** [comment]
The test failures might be a clue as to a change.

For the most part it's a case of `this string\n` in the test being asserted as `this string` so perhaps something in the test setup that's causing a newline character.

**Comment by marksweb** [comment]
@fsbraun If you've got some time, could you take a look and see if I'm missing something with the tests?

File: djangocms_alias/test_utils/text/templates/text/text.html [hunk 2]
```diff
@@ -1 +1 @@
-{{ instance.body|safe }}
\ No newline at end of file
+{{ instance.body|safe }}
```
Comment (fsbraun) [inline djangocms_alias/test_utils/text/templates/text/text.html:1]: Here, a newline has been added which must be removed. Sometimes editors do this. I have this issue with PyCharm, for example.

Comment (fsbraun) [reply djangocms_alias/test_utils/text/templates/text/text.html:1]: Ah, there we go. I'll also have to do this for djanogcms-text[-ckeditor]...

**Review by marksweb — APPROVED** [review]

**Pull request merged**